manifests/init.pp

```diff
@@ -898,24 +906,28 @@
     key          => $key,
   }
 
-  file { 'ssh_known_hosts':
-    ensure => file,
-    path   => $ssh_config_global_known_hosts_file,
-    owner  => $ssh_config_global_known_hosts_owner,
-    group  => $ssh_config_global_known_hosts_group,
-    mode   => $ssh_config_global_known_hosts_mode,
-  }
+  if $manage_global_known_hosts_real == true {
 
-  # import all nodes' ssh keys
-  if $ssh_key_import_real == true {
-    Sshkey <<||>> {
-      target => $ssh_config_global_known_hosts_file,
+    file { 'ssh_known_hosts':
+      ensure => file,
+      path   => $ssh_config_global_known_hosts_file,
+      owner  => $ssh_config_global_known_hosts_owner,
+      group  => $ssh_config_global_known_hosts_group,
+      mode   => $ssh_config_global_known_hosts_mode,
+    }
+
+    # import all nodes' ssh keys
+    if $ssh_key_import_real == true {
+      Sshkey <<||>> {
+        target => $ssh_config_global_known_hosts_file,
+      }
+    }
+
+    # remove ssh key's not managed by puppet
+    resources  { 'sshkey':
+      purge => $purge_keys_real,
     }
-  }
 
-  # remove ssh key's not managed by puppet
-  resources  { 'sshkey':
-    purge => $purge_keys_real,
   }
```

Comment:
This is trivial, but since you are already rebasing changes for the tweaks to the spec tests, could you please remove the empty line between closing brackets.

So instead of

```puppet
  # remove ssh key's not managed by puppet
  resources  { 'sshkey':
    purge => $purge_keys_real,
  }

}
```

put

```puppet
  # remove ssh key's not managed by puppet
  resources  { 'sshkey':
    purge => $purge_keys_real,
  }
}
```